graylog2-server/src/main/java/org/graylog2/indexer/IndexHelper.java

```diff
@@ -83,9 +86,14 @@ public static QueryBuilder getTimestampRangeFilter(TimeRange range) throws Inval
     }
 
     public static Set<IndexRange> determineAffectedIndicesWithRanges(IndexRangeService indexRangeService,
-                                                                     TimeRange range) {
+                                                                     TimeRange range,
+                                                                     String indexPrefix) {
         final ImmutableSortedSet.Builder<IndexRange> indices = ImmutableSortedSet.orderedBy(IndexRange.COMPARATOR);
         for (IndexRange indexRange : indexRangeService.find(range.getFrom(), range.getTo())) {
+            // if we only consider a certain index set, filter the index ranges by it.
+            if (indexPrefix != null && !indexRange.indexName().startsWith(indexPrefix + "_")) {
```

Comment:
We should use `IndexSet#isManagedIndex()` here because that is more strict and will be even more strict in the future.